src/java/src/main/java/ai/onnxruntime/genai/SimpleGenAI.java

```diff
@@ -0,0 +1,132 @@
+ * class as the `listener` argument.
+ */
+public class SimpleGenAI {
+  private Model model;
+  private Tokenizer tokenizer;
+
+  public SimpleGenAI(String modelPath) throws GenAIException {
+    model = new Model(modelPath);
+    tokenizer = new Tokenizer(model);
+  }
+
+  /**
+   * This interface represents a listener for token updates. When a new token is generated, the
+   * listener will be called with the token.
+   *
+   * <p>WARNING: Generation of the next token will be blocked until the listener returns.
+   */
+  // public interface TokenUpdateListener {
+  //   /**
+  //    * Called when a new token is generated.
+  //    *
+  //    * @param token The new token.
+  //    */
+  //   void onTokenGenerate(String token);
+  // }
+
+  /**
+   * Create the generator parameters and add the prompt text. The user can set other search options
+   * via the GeneratorParams object prior to running `generate`.
+   *
+   * @param prompt The prompt text to encode.
+   * @return The generator parameters.
+   * @throws GenAIException on failure
+   */
+  GeneratorParams createGeneratorParams(String prompt) throws GenAIException {
+    GeneratorParams generatorParams = model.createGeneratorParams();
+
+    try (Sequences encodedPrompt = tokenizer.encode(prompt)) {
+      generatorParams.setInput(encodedPrompt);
+    } catch (GenAIException e) {
+      generatorParams.close();
+      throw e;
+    }
+
+    return generatorParams;
+  }
+
+  /**
+   * Create the generator parameters and add the prompt text. The user can set other search options
+   * via the GeneratorParams object prior to running `generate`.
+   *
+   * @return The generator parameters.
+   * @throws GenAIException on failure
+   */
+  GeneratorParams createGeneratorParams(int[] tokenIds, int sequenceLength, int batchSize)
+      throws GenAIException {
+    GeneratorParams generatorParams = model.createGeneratorParams();
+    try {
+      generatorParams.setInput(tokenIds, sequenceLength, batchSize);
+    } catch (GenAIException e) {
+      generatorParams.close();
+      throw e;
+    }
+
+    return generatorParams;
+  }
+
+  /**
+   * Generate text based on the prompt and settings in GeneratorParams.
+   *
+   * <p>NOTE: This only handles a single sequence of input (i.e. a single prompt which equates to
+   * batch size of 1)
+   *
+   * @param generatorParams The prompt and settings to run the model with.
+   * @param listener Optional callback for tokens to be provided as they are generated. NOTE: Token
+   *     generation will be blocked until the listener's `accept` method returns.
+   * @return The generated text.
+   * @throws GenAIException on failure
+   */
+  public String generate(GeneratorParams generatorParams, Consumer<String> listener)
+      throws GenAIException {
+    String result = null;
+    try (Tokenizer tokenizer = new Tokenizer(model)) {
+      int[] output_ids = null;
```

Comment:
Both result and output_ids don't need to be set to null, the Java compiler knows they are assigned if the method terminates normally.